engine/rpcserver.go

```diff
@@ -5884,3 +5884,13 @@ func (s *RPCServer) GetCurrencyTradeURL(ctx context.Context, r *gctrpc.GetCurren
 		Url: url,
 	}, nil
 }
+
+// rpcContextToLongLivedSession converts a short-lived incoming context to a long-lived outgoing context, this is due
+// to the incoming context being cancelled when the RPC call completes.
+func (s *RPCServer) rpcContextToLongLivedSession(ctx context.Context) context.Context {
```

Comment:
🐒 Optional nitpick:
Okay; I kinda get it, except `LongLivedSession` seems ... well weird.
We're basically just stripping out the lifecycle from the context, right?
So ... isn't this just a very fancy `context.WithoutCancel` ?

Reply:
Yeah good point, any other naming suggestions and I will change it, and should this function live somewhere like common?

Reply:
Well I feel like an idiot. 😆